src/cli/javascript/request_filter.rs

```diff
@@ -65,10 +75,11 @@ impl RequestFilter {
 
 #[async_trait::async_trait]
 impl HttpIO for RequestFilter {
-    async fn execute(
-        &self,
+    async fn execute_with<'a>(
+        &'a self,
         request: reqwest::Request,
+        http_filter: &'a Option<http::HttpFilter>,
     ) -> anyhow::Result<Response<hyper::body::Bytes>> {
-        self.on_request(request).await
+        self.on_request(request, http_filter).await
```

Comment:
Need to check if filter.on_request is defined. If not, we should log a warning and simly call client.execute.

Reply:
That i am handling in on_request.
should we need here?

Reply:
let mut command = None;
        if let Some(value) = http_filter {
            if let Some(ref value) = value.on_request {
                command = self.worker.call(value.clone(), event).await?;
            }
        }

Reply:
ideally need to check before we call on_request here itself. will change.

Reply:
should we pass on_request to on_request fn

Reply:
@coderabbitai

Reply:
async fn execute_with<'a>(
        &'a self,
        request: reqwest::Request,
        http_filter: &'a http::HttpFilter,
    ) -> anyhow::Result<Response<hyper::body::Bytes>> {
        if Some(http_filter.on_request) != None {
            self.on_request(request, http_filter.clone()).await
        }
        else{
            self.client.execute(request).await
        }
    }

Reply:
http_filter if default
http::HttpFilter::Default()
then on_reques is None

Reply:
@coderabbitai

Reply:
@tusharmath 
please the correct structure of
#[derive(Default, Clone, Debug)]
/// User can configure the filter/interceptor
/// for the http requests.
pub struct HttpFilter {
    pub on_request: Option<String>,
}